### ISSUE TITLE
create tests that extend rspec classes instead of using describe

### ISSUE THREAD

**Issue by brianmario** [rspec/rspec-core#427]
This may already be possible, but I haven't been able to get it working correctly.

I tried following https://gist.github.com/921590 with some slight variation - instead of clobbering `Test::Unit::TestCase` I just tried to extend `RSpec::Core::ExampleGroup` with my test class, like you normally do with `Test::Unit`.

Ideally I'd like to have my own classes for units, functionals and integration tests that would extend/mixin what they need to from RSpec classes and modules to allow the exact same behavior as using `describe`.

Since I'm terrible at explaining things, here's what I'm trying to do in code:

``` ruby
class SomeIntegrationTest < SomeRspec::Request::BaseClass
  before :all do
    # ...
  end

  it "tests something" do
    # ...
  end

  context "some context" do
    before :each do
      # ...
    end

    it "tests something else" do
      # ...
    end
  end
end
```

Basically, to allow me to extend RSpec classes instead of using describe with no changes to the test code inside as if I were still using describe.

This would allow me to setup my own versions of the various RSpec base classes where I'd mixin custom functionality without the need to reopen the RSpec classes themselves.

As I said, I'm not actually sure if this is already possible but I'd be a huge help in my mission to migrate more tests away from Test::Unit to rspec2.
